lua/harpoon/list.lua

```diff
@@ -353,6 +360,11 @@ function HarpoonList.decode(list_config, name, items)
     local list_items = {}
 
     for _, item in ipairs(items) do
+        -- this is here to prevent a breaking change by converting saved relative paths to absolute. remove after some time
+        local decodedItem = vim.json.decode(item)
+        decodedItem.value = vim.loop.fs_realpath(decodedItem.value)
+        item = vim.json.encode(decodedItem)
```

Comment:
This isn't the way.  move this extra logic into the `list_decode` section.

we don't know how the list items are formed or what they look like or if they are even json.  The _default_ encode/decode does.  So rely on `decode` to do this lifting